dm/dm/config/security.go

```diff
@@ -72,6 +75,36 @@ func (s *Security) LoadTLSContent() error {
 	return nil
 }
 
+// DumpTLSContent dump tls certs data to file.
+// note that when path in security is empty, we dump certs file to subtask dump dir.
+func (s *Security) DumpTLSContent(baseDirPath string) error {
+	if s.SSLCA == "" {
+		s.SSLCA = path.Join(baseDirPath, "ca.pem")
+	}
+	if s.SSLCert == "" {
+		s.SSLCert = path.Join(baseDirPath, "cert.pem")
+	}
+	if s.SSLKey == "" {
+		s.SSLKey = path.Join(baseDirPath, "key.pem")
+	}
+	if !utils.IsFileExists(s.SSLCA) {
+		if err := utils.WriteFileAtomic(s.SSLCA, s.SSLCABytes, 0o600); err != nil {
+			return err
+		}
+	}
+	if !utils.IsFileExists(s.SSLCert) {
+		if err := utils.WriteFileAtomic(s.SSLCert, s.SSLCertBytes, 0o600); err != nil {
+			return err
+		}
+	}
+	if !utils.IsFileExists(s.SSLKey) {
```

Comment:
overwriting old content is acceptable? maybe user change the cert content

Reply:
I don't think it's a good practice to overwrite a user's file, ex: if some cert stores in mysql's dir

Reply:
MySQL's directory is located in upstream, this is DM-worker. BTW, we'll remove whole folder after finishing load unit.

Reply:
we can operate files in dm-worker's directory but i don't think user will put cert files to it ? so i perfer to if the file user configed in yaml is wrong, just return the error

Reply:
Now I think the behavior is that 
- if user specified the path for certs
   - if the cert exists, don't modify it
   - if the cert doesn't exist, we can overwrite certs to dump folder and change the cert path. Because the original user specified path maybe not suitable for this DM-worker
- if user didn't specify the path for certs, overwrite certs to dump folder and change the cert path.

And why can't we change lightning to pass cert content?

Reply:
> if the cert doesn't exist, we can overwrite certs to dump folder and change the cert path. Because the original user specified path maybe not suitable for this DM-worker

i will impl this in next commit, and all the other points mentioned above have been implemented in current code


> And why can't we change lightning to pass cert content?

as many of the components lighting relies on (tidb,tikv-client,pd-clinet) do not support the init tls.Config directly by raw tls data, dm-worker downloads the certificate file from dm-master and saves it to the specified path (already mentioned in PR description )

it's too hard  to change so many components 😢

Reply:
implemented in https://github.com/pingcap/ticdc/pull/3260/commits/a71915c9f0eac266d005c77a8a259da3addf14b2
ptal
 i will merge this pr if no more comments